compiler/src/test/java/com/squareup/anvil/compiler/codegen/BindingModuleMultibindingMapTest.kt

```diff
@@ -0,0 +1,295 @@
+    @JvmStatic fun annotationClasses(): Collection<Any> {
+      return listOf(MergeComponent::class, MergeSubcomponent::class, MergeModules::class)
+    }
+  }
+
+  @Test fun `the Dagger multibinding method is generated for non-objects`() {
+    compile(
+      """
+      package com.squareup.test
+
+      import com.squareup.anvil.annotations.ContributesMultibinding
+      import dagger.MapKey
+      $import
+      
+      @MapKey
+      annotation class BindingKey(val value: String)
+
+      interface ParentInterface
+
+      interface Middle : ParentInterface
+
+      @ContributesMultibinding(Any::class, ParentInterface::class)
+      @BindingKey("abc")
+      interface ContributingInterface : Middle
+
+      $annotation(Any::class)
+      interface ComponentInterface
+      """
+    ) {
+      val modules = if (annotationClass == MergeModules::class) {
+        componentInterface.daggerModule.includes.toList()
+      } else {
+        componentInterface.anyDaggerComponent.modules
+      }
+
+      val methods = modules.single().java.declaredMethods
+      assertThat(methods).hasLength(1)
+
+      with(methods[0]) {
+        assertThat(returnType).isEqualTo(parentInterface)
+        assertThat(parameterTypes.toList()).containsExactly(contributingInterface)
+        assertThat(isAbstract).isTrue()
+        assertThat(isAnnotationPresent(Binds::class.java)).isTrue()
+        assertThat(isAnnotationPresent(IntoMap::class.java)).isTrue()
+        assertThat(isAnnotationPresent(IntoSet::class.java)).isFalse()
+        assertThat(isAnnotationPresent(bindingKey)).isTrue()
+        assertThat(getAnnotation(bindingKey).getValue<String>()).isEqualTo("abc")
+      }
+    }
+  }
+
+  @Test fun `the Dagger provider method is generated for objects`() {
+    compile(
+      """
+      package com.squareup.test
+
+      import com.squareup.anvil.annotations.ContributesMultibinding
+      import dagger.MapKey
+      $import
+      
+      @MapKey
+      annotation class BindingKey(val value: String)
+
+      interface ParentInterface
+
+      @ContributesMultibinding(Any::class)
+      @BindingKey("abc")
+      object ContributingInterface : ParentInterface
+
+      $annotation(Any::class)
+      interface ComponentInterface
+      """
+    ) {
+      val modules = if (annotationClass == MergeModules::class) {
+        componentInterface.daggerModule.includes.toList()
+      } else {
+        componentInterface.anyDaggerComponent.modules
+      }
+      assertThat(modules).containsExactly(componentInterfaceAnvilModule.kotlin)
+
+      val methods = modules.single().java.declaredMethods
+      assertThat(methods).hasLength(1)
+
+      with(methods[0]) {
+        assertThat(returnType).isEqualTo(parentInterface)
+        assertThat(parameterTypes.toList()).isEmpty()
+        assertThat(isAbstract).isFalse()
+        assertThat(isAnnotationPresent(Provides::class.java)).isTrue()
+        assertThat(isAnnotationPresent(IntoMap::class.java)).isTrue()
```

Comment:
Redundant, but add assertThat(isAnnotationPresent(IntoSet::class.java)).isFalse() for consistency?

Reply:
That's the bad thing about the tests, there's a lot of repetition. If one condition fails, then many tests start to fail. Ideally, there would be only one failing :-/